metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/samples/prometheus/collectorRegistry/SimpleCollectorRegistryPrometheusInstanceSamplesProvider.java

```diff
@@ -0,0 +1,143 @@
+     *                   and to be prepended to each MetricFamilySample.Sample.name when building PrometheusSample.name.
+     *                   Optional
+     *
+     * @param metricFamilyNameFilter optional
+     * @param metricFamilySampleNameFilter optional
+     * @param collectorRegistries must be not empty
+     */
+    public SimpleCollectorRegistryPrometheusInstanceSamplesProvider(
+        MetricName namePrefix,
+        Predicate<String> metricFamilyNameFilter,
+        Predicate<String> metricFamilySampleNameFilter,
+        Collection<? extends CollectorRegistry> collectorRegistries) {
+
+        this.namePrefix = namePrefix;
+        this.metricFamilyNameFilter = metricFamilyNameFilter;
+        this.metricFamilySampleNameFilter = metricFamilySampleNameFilter;
+
+        checkArgument(
+            collectorRegistries != null && !collectorRegistries.isEmpty(),
+            "collectorRegistries is null or empty");
+
+        this.collectorRegistries = collectorRegistries;
+    }
+
+    @Override
+    public Set<PrometheusInstanceSample> instanceSamples() {
+        Set<PrometheusInstanceSample> result = new LinkedHashSet<>();
+        collectorRegistries.forEach(collectorRegistry -> processCollectorRegistry(collectorRegistry, result));
+        return result;
+    }
+
+    private void processCollectorRegistry(CollectorRegistry collectorRegistry, Set<PrometheusInstanceSample> result) {
+        Enumeration<MetricFamilySamples> fsEnumeration =
+            metricFamilyNameFilter != null ?
+            collectorRegistry.filteredMetricFamilySamples(metricFamilyNameFilter) :
+            collectorRegistry.metricFamilySamples();
+
+        while (fsEnumeration.hasMoreElements()) {
+            MetricFamilySamples fs = fsEnumeration.nextElement();
+
+            MetricName name =
+                hasNamePrefix() ?
+                MetricName.of(namePrefix, fs.name) :
+                MetricName.of(fs.name);
+
+            PrometheusInstanceSample instanceSample = new PrometheusInstanceSample(name, name, fs.help, fs.type);
+
+            fs.samples.forEach(fsSample -> {
+                if (metricFamilySampleNameFilter != null && !metricFamilySampleNameFilter.test(fsSample.name)) {
+                    return;
+                }
+
+                MetricName sampleName =
+                    hasNamePrefix() ?
+                    MetricName.of(namePrefix, fsSample.name) :
+                    MetricName.of(fsSample.name);
+
+                PrometheusSample sample = new PrometheusSample(
+                    null,
+                    null,
+                    sampleName,
+                    null,
+                    fsSample.labelNames,
+                    fsSample.labelValues,
+                    fsSample.value);
+
+                instanceSample.add(sample);
+            });
+
+            if (instanceSample.samples() != null && !instanceSample.samples().isEmpty()) {
```

Comment:
It can't be null - we don't need this check.

Reply:
+1, will fix it

Reply:
Fixed